src/components/OcCheckbox.vue

```diff
@@ -2,15 +2,17 @@
   <label :for="id" :class="{ 'oc-cursor-pointer': !disabled }">
     <input
       :id="id"
-      v-model="$_ocCheckbox_model"
+      v-model="model"
       type="checkbox"
       name="checkbox"
-      :aria-label="label"
-      :class="$_ocCheckbox_classes"
+      :aria-labelledby="labelId"
+      :aria-checked="checked.toString()"
+      :aria-disabled="disabled.toString()"
+      :class="classes"
       :value="option"
       :disabled="disabled"
     />
-    <span v-if="!hideLabel" :aria-hidden="true" v-text="label" />
+    <span v-if="!hideLabel" :id="labelId" :aria-hidden="true" v-text="label" />
```

Comment:
This is a bit more complicated. There are valid scenarios for both having a visual label and having no visual label. But in any scenario the checkbox needs to have an accessible label. I'd propose the following:
- get rid of the `hideLabel` prop
- keep the `label` prop (optional)
- add an `accessibleLabel` prop (required)
=> then have two spans: the one for `label` can be aria-hidden, the one for the `accessibleLabel` needs to be invisible for the user (use `oc-invisible-sr` class), but is the one actually used for the `aria-labelledby` prop on the checkbox. There will be situations then where the provided values for `label` and `accessibleLabel` are the same. I'm fine with that to not get an overcomplicated logic here.

Reply:
I don't like the idea:
then we have the top label and 2 spans inside, will come up with a better solution

Reply:
Maybe I am missing something, but why not just remove `:aria-hidden="true"` from the current implementation? Then you can hide the label or not, but regardless it is used to label the checkbox (and therefore required ofc).

Reply:
have a look